src/org/ods/component/ScanWithSnykStage.groovy

```diff
@@ -20,19 +20,29 @@ class ScanWithSnykStage extends Stage {
     if (!config.buildFile) {
       config.buildFile = 'build.gradle'
     }
+    if (!config.severityThreshold) {
+      // low is the default, it is equal to not providing the option to snyk
+      config.severityThreshold = 'low'
+    } else {
+      config.severityThreshold = config.severityThreshold.trim().toLowerCase()
+    }
     this.snyk = snyk
   }
 
   def run() {
     if (!config.snykAuthenticationCode) {
       script.error "Option 'snykAuthenticationCode' is not set!"
     }
+    if(!config.severityThreshold.matches('\\b^(low|medium|high)$\\b')) {
+      script.error "'${config.severityThreshold}' is not a valid value for option 'severityThreshold'! Please use low, medium or high."
+    }
```

Comment:
```suggestion
def allowedSeverityThresholds = ['low', 'medium', 'high']
if(!allowedSeverityThresholds.contains(config.severityThreshold)) {
       script.error "'${config.severityThreshold}' is not a valid value for option 'severityThreshold'! Please use one of ${allowedSeverityThresholds}."
     }
```

Reply:
In general, not sure why I did the check for `snykAuthenticationCode` here instead of the constructor. Might be better to check all config options there?

Reply:
I first had it in the constructor but then you might not find easily in the logs in which stage this error relates too since it happened before the stage start output.